UI/DesignWnd.cpp

```diff
@@ -210,24 +212,20 @@ namespace {
     SavedDesignsManager& GetSavedDesignsManager()
     { return SavedDesignsManager::GetSavedDesignsManager(); }
 
-    void WriteDesignToFile(int design_id, boost::filesystem::path& file) {
-        const ShipDesign* design = GetShipDesign(design_id);
-        if (!design)
-            return;
-
+    void WriteDesignToFile(const ShipDesign& design, boost::filesystem::path& file) {
```

Comment:
I added a TraceLogger statement.

I did not add a messages window message.  The user confirmation that the save worked is that the saved design changed in the design window.

From an in game perspective these saves are not a user initiated disk action, and should not be confirmed in game.

Saving the game is user initiated and there is no other confirmation (spinner etc.) that the save succeeded.  The user initiates the save and may expect a confirmation that the save has happened.

Adjusting the tech status filter also changes the on disk "config.xml" file but does not have a message window confirmation of the change.  The confirmation that it worked is immediate change of the filter status.  The confirmation that it persisted is when it re-loads in the same state that it exited with.

During development I did not have any additional debug statements in this location that I subsequently removed.  So I think more logging here just adds noise.